File: app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.utils.enums import NotificationChannel


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    channel: NotificationChannel
    title: str
    message: str
    payload: dict
    read_at: datetime | None
    created_at: datetime

File: eldas-frontend/src/types/notification.ts
export type NotificationChannel = "in_app" | "email";

export interface NotificationItem {
  id: number;
  user_id: number;
  channel: NotificationChannel;
  title: string;
  message: string;
  payload: Record<string, unknown>;
  read_at: string | null;
  created_at: string;
}
